### PR TITLE
Eliminate Example- YAML's and update build-yaml files for user usage

### PR DESCRIPTION
**Pull request by barndawgie**
Apologies for this slightly larger change; there are a few separate but related items in here:

- Update the YAML files under `build-yaml/` so they can be used by new users in place of the existing `example-*.yaml` files
- Update the README to refer to these files
- Simplify and align all build scripts and update them to handle the changes to these files
- Eliminate the `example-*.yaml` files which should no longer be needed

I plan follow-on PR's to:

- Update README to refer to Web Tools install and move much of the current material to the wiki
- Update Github Actions workflows to re-use the build workflow for the other workflows (to avoid code duplication that is there now that this change has aligned the workflows to all work the same)

### CODE REVIEW

**Comment by barndawgie** [comment]
One other note - I did test the build and publish workflow as well:
https://github.com/barndawgie/esphome-econet/actions/runs/6866743645

Results are available at:
https://barndawgie.github.io/install/

File: README.md [hunk 1]
```diff
@@ -58,22 +58,20 @@ Installation is a three-step process:
 
 ### Step 1: Customizing the esphome-econet yaml for Your Device
 
-Before installing, you will need to create a customized yaml for your device/environment. Start with either the [example-esp32.yaml](https://github.com/stockmopar/esphome-econet/blob/main/example-esp32.yaml) or [example-esp8266.yaml](https://github.com/stockmopar/esphome-econet/blob/main/example-esp8266.yaml) files as a template, depending on your chosen hardware.
+Before installing, you will need to create a customized yaml for your device/environment. Start with one of the starter yaml's in in the [build-yaml directory](https://github.com/esphome-econet/esphome-econet/tree/main/build-yaml) of the repo, choosing the appropriate file for your appliance and ESP hardware (choose ESP32 for the standard ATOM Lite hareware described above):
```
[the remark this answers is not in the document]
Comment (barndawgie) [reply README.md]: Good catch. Not sure why the spellchecker missed that one.

[the remark this answers is not in the document]
Comment (barndawgie) [reply README.md]: https://github.com/codespell-project/codespell/pull/3215

File: build-yaml/econet-etwh-esp32.yaml [hunk 2]
```diff
@@ -6,8 +6,16 @@ substitutions:
   board: esp32dev
 
 packages:
-  econet-etwh: !include ../econet_electric_tank_water_heater.yaml
+  econet:
+    url: https://github.com/esphome-econet/esphome-econet
+    ref: main
```
Comment (tronikos) [inline build-yaml/econet-etwh-esp32.yaml:11]: Use github_ref substitution? Same elsewhere.

File: build-yaml/econet-hvac-esp32.yaml [hunk 3]
```diff
@@ -6,10 +6,18 @@ substitutions:
   board: esp32dev
 
 packages:
-  econet-hvac: !include ../econet_hvac.yaml
+  econet:
+    url: https://github.com/esphome-econet/esphome-econet
+    ref: main
+    file: econet_hvac.yaml
   # Uncomment if you have an communicationg Outdoor Unit
   # econet-hvac-odu: !include ../econet_hvac_odu.yaml
```
Comment (tronikos) [inline build-yaml/econet-hvac-esp32.yaml:14]: Update this too. Same in the the other file.

File: build-yaml/econet-etwh-esp32.yaml (hunk 2 above)
Comment (barndawgie) [reply build-yaml/econet-etwh-esp32.yaml:11]: Unfortunately you can't use substitutions for `packages:` as per https://github.com/esphome/issues/issues/4939

File: build-yaml/econet-hvac-esp32.yaml (hunk 3 above)
Comment (barndawgie) [reply build-yaml/econet-hvac-esp32.yaml:14]: Unfortunately you can't use substitutions for `packages:` as per https://github.com/esphome/issues/issues/4939

Comment (tronikos) [reply build-yaml/econet-hvac-esp32.yaml:14]: I meant update this too to include from the repository URL instead of a relative path.

Comment (barndawgie) [reply build-yaml/econet-hvac-esp32.yaml]: Oh, I see. I thought this was the same comment as above. Good catch! Updated.

**Review by tronikos — APPROVED** [review]

**Pull request merged**